Add cudf snowflake example

ncclementi opened the pull request:
Closes https://github.com/rapidsai/deployment/issues/494 

With the caveat of having to do this:

```python
from cudf.pandas.module_accelerator import disable_module_accelerator

with disable_module_accelerator():
    df = table.to_pandas()

import pandas as pd

df = pd.DataFrame(df)  # this will take a few seconds
```

jacobtomlinson commented on source/examples/rapids-snowflake-cudf/notebook.ipynb:
```diff
@@ -0,0 +1,356 @@
+{
```
<p>I would use the <a href="https://github.com/rapidsai/deployment?tab=readme-ov-file#docref" target="_blank">docref</a> admonition int he first paragraph to cross reference the Snowflake page. This has a few benefits including inserting a link back to this example at the bottom of the snowflake page.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

jacobtomlinson replied:
<pre class="ql-syntax" spellcheck="false">Line #14.        "token": get_login_token(),
</pre><p>You could replace this line with <code>"token": pathlib.Path("/snowflake/session/token").read_text()</code> which would remove a little indirection while still being very readable.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

jacobtomlinson replied:
<p>I would make this note a warning admonition. We might also want to link to the issue so that readers can check on the progress.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

jacobtomlinson replied:
<p>I would use the <a href="https://github.com/rapidsai/deployment?tab=readme-ov-file#docref" target="_blank">docref</a> admonition int he first paragraph to cross reference the Snowflake page. This has a few benefits including inserting a link back to this example at the bottom of the snowflake page.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

jacobtomlinson replied:
<pre class="ql-syntax" spellcheck="false">Line #14.        "token": get_login_token(),
</pre><p>You could replace this line with <code>"token": pathlib.Path("/snowflake/session/token").read_text()</code> which would remove a little indirection while still being very readable.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

jacobtomlinson replied:
<p>I would make this note a warning admonition. We might also want to link to the issue so that readers can check on the progress.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

jacobtomlinson replied:
<p>I would use the <a href="https://github.com/rapidsai/deployment?tab=readme-ov-file#docref" target="_blank">docref</a> admonition int he first paragraph to cross reference the Snowflake page. This has a few benefits including inserting a link back to this example at the bottom of the snowflake page.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

jacobtomlinson replied:
<pre class="ql-syntax" spellcheck="false">Line #14.        "token": get_login_token(),
</pre><p>You could replace this line with <code>"token": pathlib.Path("/snowflake/session/token").read_text()</code> which would remove a little indirection while still being very readable.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

jacobtomlinson replied:
<p>I would make this note a warning admonition. We might also want to link to the issue so that readers can check on the progress.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

jacobtomlinson commented:
Looks like ReviewNB had an issue and created some duplicated comments. I'll clean those up now.

ncclementi commented:
We have an issue, I was trying to test the using `Path` to replace `get_login_token()` and now I can't seem to be able to stablish a session. It just hangs in 
```
session = Session.builder.configs(connection_parameters).create()
```

it's unclear to me why, I did turn of MFA just in case, I wonder if there is some token caching or something that might be causing a problem.

ncclementi commented:
I went through the deployment, and then connected and things worked. I'm not sure what was the hiccup yesterday but clean slate worked. 

I updated the example according to @jacobtomlinson suggestions, and rebased against main to update the branch. This should be ready for final review.

jacobtomlinson reviewed (approved):
Awesome this looks great!
